### PR TITLE
Hinted handoff throttling

### PR DESCRIPTION
**Pull request by jwilder**
This adds some hinted handoff queue processing rate limits to reduce the rate of writes sent to nodes coming back online.

Also adds some logging around the processing of the queues and more context when writes fail.

### CODE REVIEW

**Comment by otoolep** [comment]
The limiter is pretty important to get right. Can it be unit-tested? Even in a basic manner?

File: services/hh/limiter.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	t.count += int64(count)
+}
+
+// Wait will block a variable amount of time, up to 1 second, to maintain the configured rate
+func (t *limiter) Wait() {
+	if t.limit > 0 {
+		delta := time.Now().Sub(t.start).Seconds()
+		rate := int64(float64(t.count) / delta)
+
+		maxRate := t.limit
```
Comment (otoolep) [inline services/hh/limiter.go]: Why have this variable? Purely cosmetic?

**Comment by otoolep** [comment]
OK, this seems to make sense. The idea is that the system delays, up to a second, in proportion to how much the actual rate is above the desired rate. 

I would like to see some sort of testing around the limiter though. If it's wrong, or regresses, it could have a serious impact on the system. Thoughts?

File: services/hh/processor.go [hunk 2]
```diff
@@ -122,6 +123,17 @@ func (p *Processor) Process() error {
...
+			// Log how many writes we successfully sent at the end
+			var sent int
+			start := time.Now()
+			defer func(start time.Time) {
+				if sent > 0 {
+					p.Logger.Printf("%d queued writes sent to node %d in %s", sent, nodeID, time.Since(start))
+				}
+			}(start)
+
+			limiter := NewRateLimiter(0)
```
Comment (otoolep) [inline services/hh/processor.go]: This will eventually need to be set through a config variable, right?

Comment (jwilder) [inline services/hh/processor.go]: Fixed.  There's already a config var for it.

**Comment by jwilder** [comment]
Added some tests.

**Pull request merged**